include/translated-term.php

```diff
@@ -338,4 +338,81 @@ protected function get_db_infos() {
 			'default_alias' => 't',
 		);
 	}
+
+	/**
+	 * Wrapper of `wp_insert_term` with the language.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string       $term     The term name to add.
+	 * @param string       $taxonomy The taxonomy to which to add the term.
+	 * @param PLL_Language $language The term language.
+	 * @param array|string $args     The taxonomy of the term.
+	 * @return array|WP_Error An array of the new term, WP_Error otherwise.
+	 *
+	 * @phpstan-param array{
+	 *   alias_of?: string,
+	 *   description?: string,
+	 *   parent?: int,
+	 *   slug?: string,
+	 * } $args
+	 */
+	public function insert_term( string $term, string $taxonomy, PLL_Language $language, $args = array() ) {
+		$set_language_for_term_slug = function () use ( $language ) {
+			return $language;
+		};
+
+		add_filter( 'pll_inserted_term_language', $set_language_for_term_slug, 20 ); // After Polylang's filter.
```

Comment:
We should also filter `pll_insterted_term_parent`. Indeed, if there is one it can be found in `$args['parent']`.